src/components/AddBook.jsx

```diff
@@ -1,21 +1,19 @@
 import { useState } from 'react';
-import { useDispatch, useSelector } from "react-redux";
-import { addBook } from "../redux/books/booksSlice";
+import { useDispatch } from "react-redux";
+import { addNewBook, fetchBooks } from '../redux/books/booksSlice';
 import Button from './Button';
 
 const AddBook = () => {
 
   const dispatch = useDispatch();
   const [title, setTitle] = useState("");
   const [author, setAuthor] = useState("");
-  const books = useSelector((state) => state.books.books)
 
   const submitForm = (e) => {
     e.preventDefault();
 
-    const id = books.length + 1
-
-    dispatch(addBook({ item_id: `item${id}`, title, author }));
+    dispatch(addNewBook({ title, author, category: "" }));
+    dispatch(fetchBooks())
   }
```

Comment:
- [ ] Kindly make sure that the app works properly. Whenever a task is created or removed the whole list disappears from the DOM. kindly fix this.